### DISCUSSION TITLE
Could we prepare a new release?

### DISCUSSION THREAD

**Discussion by rubenporras** [eclipse/lsp4e#760] [General]
Could we prepare a new LSP4E release?

The reason is that recently we upgraded to Eclipse 2023-06 and now we wanted to upgrade to LSP4E 2.23, but we have hit the bug https://github.com/eclipse/lsp4e/commit/21d3a1775a5c874eccc22812a47ee506e2844cb7, which is fixed now on master. It looks like in combination with the change done https://github.com/eclipse/lsp4e/pull/554, the bug would hit us quite often, preventing us from upgrading LSP4E.

**Comment by angelozerr** [comment]
> we have hit the bug https://github.com/eclipse/lsp4e/commit/21d3a1775a5c874eccc22812a47ee506e2844cb7

I give you just this information, but I have no usecases in LSP4E. I tried to apply this patch in LSP4IJ and I had sometimes deadlock because of synchronized use. Perhaps in Eclipse IDE the Thread context is different and you have not the same problem than IJ, but I wanted to share my problem with this fix in IJ context.

**Comment by mickaelistria** [comment]
A release is (almost) always welcome.
@rubenporras would you feel ok to lead and perform this release? I don't remember we have a documentation about that, but it's mostly a matter of running the promotion job, tag (if job doesn't do it) and announce.

**Comment by rubenporras** [comment]
Yes, I can do that.

It looks like the promotion job creates the tag. I will run it later today with version 0.24 unless someone disagrees. 

I am not sure how to do the part about the announcement, do I need some right on the project to be able to do that?

**Reply by mickaelistria** [reply]
> I am not sure how to do the part about the announcement, do I need some right on the project to be able to do that?

So far, it was mostly a matter of sending an email to the lsp4e-dev mailing-list. But you may prefer to create a GitHub release and post about it as a GitHub discussion. There is no important rule about to announce, so it's up to the one(s) who do it to do it as they think is best.

**Comment by rubenporras** [comment]
The job has run and the plugins are available in https://download.eclipse.org/lsp4e/releases/0.24.0/plugins/, however, the main GitHub page still shows the old release

![image](https://github.com/eclipse/lsp4e/assets/43636626/2b512135-4c98-4c7e-b942-22a06bc0a374)

is this updated automatically or do I need to do something?

**Reply by mickaelistria** [reply]
So far, I've always created GitHub releases manually.

**Comment by rubenporras** [comment]
The packaging is done and I have created the GitHub release as well. I wanted to create an announcement in https://github.com/eclipse/lsp4e/discussions/categories/announcements, but it does not work for me. 

Could you do that?

**Reply by mickaelistria** [reply]
It looks like it cannot be done automatically after the fact: https://github.com/orgs/community/discussions/3922
Can't you directly create a new announcement as a GitHub discussion?

**Comment by rubenporras** [comment]
Thanks, I have created https://github.com/eclipse/lsp4e/discussions/772. Next time I will create the discussion when creating the release.

**Comment by jonahgraham** [comment]
Some of the "paperwork" for the release(s) needs to be done. For example I have created the [0.24.0](https://projects.eclipse.org/projects/technology.lsp4e/releases/0.24.0) and [0.24.1](https://projects.eclipse.org/projects/technology.lsp4e/releases/0.24.1) PMI entries.

Also I started this discussion https://github.com/eclipse/lsp4e/discussions/790 on simrel contribution.
